### PR TITLE
Add benchmarks for parsing and evaluating style functions

### PR DESCRIPTION
**Pull request by anandthakker**
Benchmarks for "parsing" (converting from JSON) and evaluating each type of DDS function.  

Example results:

```
---------------------------------------------------------------------
Benchmark                              Time           CPU Iterations
---------------------------------------------------------------------
Parse_CameraFunction/1              5912 ns       5884 ns     118763 1
Parse_CameraFunction/2              7563 ns       7526 ns     100021 2
Parse_CameraFunction/4             10579 ns      10568 ns      65460 4
Parse_CameraFunction/6             14101 ns      14078 ns      46792 6
Parse_CameraFunction/8             18148 ns      18129 ns      38659 8
Parse_CameraFunction/10            21299 ns      21260 ns      32044 10
Parse_CameraFunction/12            25631 ns      25598 ns      27472 12
Evaluate_CameraFunction/1            105 ns        104 ns    6943067 1
Evaluate_CameraFunction/2            165 ns        165 ns    4281738 2
Evaluate_CameraFunction/4            185 ns        184 ns    3794532 4
Evaluate_CameraFunction/6            193 ns        193 ns    3544914 6
Evaluate_CameraFunction/8            198 ns        198 ns    3524282 8
Evaluate_CameraFunction/10           211 ns        211 ns    3243789 10
Evaluate_CameraFunction/12           212 ns        211 ns    3273705 12
Parse_CompositeFunction/1           9481 ns       9456 ns      73828 1
Parse_CompositeFunction/2          19743 ns      19648 ns      36176 2
Parse_CompositeFunction/4          58765 ns      58655 ns      11013 4
Parse_CompositeFunction/6         124609 ns     124454 ns       5480 6
Parse_CompositeFunction/8         219764 ns     219296 ns       3238 8
Parse_CompositeFunction/10        335272 ns     334335 ns       2009 10
Parse_CompositeFunction/12        503914 ns     503198 ns       1456 12
Evaluate_CompositeFunction/1        2727 ns       2723 ns     251411 1
Evaluate_CompositeFunction/2        3255 ns       3252 ns     209955 2
Evaluate_CompositeFunction/4        4555 ns       4552 ns     153638 4
Evaluate_CompositeFunction/6        5690 ns       5685 ns     114965 6
Evaluate_CompositeFunction/8        6955 ns       6935 ns      97806 8
Evaluate_CompositeFunction/10       8320 ns       8311 ns      75999 10
Evaluate_CompositeFunction/12       9433 ns       9423 ns      61013 12
Parse_SourceFunction/1              6724 ns       6716 ns     100830 1
Parse_SourceFunction/2              8174 ns       8161 ns      84843 2
Parse_SourceFunction/4             11436 ns      11412 ns      61432 4
Parse_SourceFunction/6             15073 ns      15045 ns      46060 6
Parse_SourceFunction/8             18658 ns      18635 ns      35066 8
Parse_SourceFunction/10            22813 ns      22715 ns      30112 10
Parse_SourceFunction/12            27464 ns      27412 ns      26248 12
Evaluate_SourceFunction/1           1094 ns       1093 ns     604569 1
Evaluate_SourceFunction/2           1199 ns       1199 ns     560852 2
Evaluate_SourceFunction/4           1212 ns       1209 ns     516251 4
Evaluate_SourceFunction/6           1258 ns       1256 ns     549274 6
Evaluate_SourceFunction/8           1224 ns       1223 ns     559186 8
Evaluate_SourceFunction/10          1247 ns       1245 ns     558592 10
Evaluate_SourceFunction/12          1275 ns       1272 ns     540119 12
```

### CODE REVIEW

**Comment by anandthakker** [comment]
@kkaefer is the `android-debug-arm-v7` CI failure due to any known issues that you're aware of?

**Comment by kkaefer** [comment]
Yeah, it's tracked in https://github.com/mapbox/mapbox-gl-native/issues/9829

File: benchmark/function/camera_function.benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    std::string stops = "[";
+    for (size_t i = 0; i < stopCount; i++) {
+        std::string value = std::to_string(24.0f / stopCount * i);
+        if (stops.size() > 1) stops += ",";
+        stops += "[" + value + ", " + value + "]";
+    }
+    stops += "]";
+    
+    doc.Parse<0>(R"({"type": "exponential", "base": 2,  "stops": )" + stops + R"(, "property": "x"})");
```
Comment (jfirebaugh) [inline benchmark/function/camera_function.benchmark.cpp]: `"property": "x"` looks like a copy and paste mistake from `source_function.benchmark.cpp`.

Comment (anandthakker) [reply benchmark/function/camera_function.benchmark.cpp]: Fixed

**Pull request merged**